File: c/StackImplementationUsingArrays.cpp
#include<iostream>
using namespace std;

#define MAX 1000

class Stack{
    int top;
    public:
    Stack(){
        top=-1;
    }
    int arr[MAX];

    void push(int x){
        if(top < MAX)
            arr[++top] = x;
        else
            cout<<"Stack Overflow!!";
    }

    int pop(){
        if(top >= 0){
            return arr[--top];
        }else{
            cout<<"Stack Underflow!!";
            return 0;
        }
    }
};

int main(){
    class Stack st;
    st.push(4);
    cout<<st.pop();
    return 0;
}

File: java/StackImplementationUsingArrays.java
class Stack {
	static final int MAX = 1000;
	int top;
	int arr[] = new int[MAX]; 
	
	Stack() {
		top = -1;
	}

	void push(int x) {
		if (top >= (MAX - 1)) {
			System.out.println("Stack Overflow");			
		}
		else {
			arr[++top] = x;
			System.out.println(x);
		}
	}

	int pop() {
		if (top < 0) {
			System.out.println("Stack Underflow");
			return 0;
		}
		else {
			int x = arr[top--];
			return x;
		}
	}

	int peek() {
		if (top < 0) {
			System.out.println("Stack Underflow");
			return 0;
		}
		else {
			int x = arr[top];
			return x;
		}
	}
}

class Main {
	public static void main(String args[]) {
		Stack st = new Stack();
		st.push(12);
		st.push(22);
		System.out.println(st.pop());
	}
}
